src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java

```diff
@@ -264,12 +253,14 @@ public static FeedSource requestFeedSourceById(Request req, Actions action) {
 
     public static FeedSource checkFeedSourcePermissions(Request req, FeedSource feedSource, Actions action) {
         Auth0UserProfile userProfile = req.attribute("user");
-        Boolean publicFilter = Boolean.valueOf(req.queryParams("public")) ||
+        boolean publicFilter = Boolean.valueOf(req.queryParams("public")) ||
                 req.url().split("/api/*/")[1].startsWith("public");
 
         // check for null feedSource
-        if (feedSource == null)
+        if (feedSource == null) {
             logMessageAndHalt(req, 400, "Feed source ID does not exist");
+            return null;
```

Comment:
Do we need to return null after a `logMessageAndHalt` now?

Reply:
Well, it silences warnings from Intellij about objects (e.g., `feedSource` in this case) being null farther down in the method body. Maybe a better approach would to give `logMessageAndHalt` a return value of null and add a `return` in front of anytime we call that method?

Reply:
Oh interesting, I'm not getting those warnings in IntelliJ, but I could see that being useful. How did you enable these warnings?

Reply:
Not exactly sure haha, but it's called "method invocation may produce nullpointerexception."